save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/HasErrorModal.kt

```diff
@@ -46,18 +47,30 @@ val requestModalHandler: FC<PropsWithChildren> = FC { props ->
                 errorMessage = "You are not logged in",
                 errorLabel = "Unauthenticated",
                 confirmationText = "Proceed to login page",
+                status = response?.status,
+            )
+        } else if (response?.status == 404.toShort()) {
+            ErrorModalState(
+                isErrorModalOpen = false,
+                errorMessage = "${response?.status} ${response?.statusText}",
+                errorLabel = response?.status.toString(),
+                status = response?.status,
             )
         } else {
             ErrorModalState(
                 isErrorModalOpen = response != null,
                 errorMessage = "${response?.status} ${response?.statusText}",
                 errorLabel = response?.status.toString(),
+                status = response?.status,
             )
         }
         setModalState(newModalState)
     }
 
     modal { modalProps ->
```

Comment:
I had in mind something more like
```kotlin
if (modalState.status == 404.toShort()) {
    Redirect { to = ... }
} else {
    modal {}
}
```
For me the current version (when location is changed inside rendering of `modal`) looks like an unexpected side effect

Reply:
Changed it